src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedRecordCopyCtor : SynthesizedInstanceConstructor
+    {
+        public SynthesizedRecordCopyCtor(
+            SourceMemberContainerTypeSymbol containingType,
+            DiagnosticBag diagnostics)
+            : base(containingType)
+        {
+            Debug.Assert(!containingType.IsStructType(), "only reference types should define copy constructors");
+            Parameters = ImmutableArray.Create(SynthesizedParameterSymbol.Create(
+                this,
+                TypeWithAnnotations.Create(
+                    isNullableEnabled: true,
+                    ContainingType),
+                ordinal: 0,
+                RefKind.None));
+        }
+
+        public override ImmutableArray<ParameterSymbol> Parameters { get; }
+
+        internal override LexicalSortKey GetLexicalSortKey()
+        {
+            // We need a separate sort key because struct records will have two synthesized
+            // constructors: the record constructor, and the parameterless constructor
+            return LexicalSortKey.SynthesizedRecordCopyCtor;
+        }
+
+        internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F, ArrayBuilder<BoundStatement> statements, DiagnosticBag diagnostics)
+        {
+            // Write assignments to backing fields
+            //
+            // {
+            //     this.backingField1 = arg1
+            //     ...
+            //     this.backingFieldN = argN
+            // }
+            var param = F.Parameter(Parameters[0]);
+            foreach (var member in ContainingType.GetMembers())
```

Comment:
>GetMembers [](start = 50, length = 10)

This fails to copy all fields of the input type. Specifically, it fails to copy inherited fields.

Reply:
I suspect this is intentional for now (we're not yet supporting inheritance from or to record type). But adding a PROTOTYPE would be good.
Also, what happens if the base type has an obsolete field, which the `Clone` will use. Should that be reported?

---
In reply to: [431496592](https://github.com/dotnet/roslyn/pull/44582#discussion_r431496592) [](ancestors = 431496592)